### PR TITLE
Hashicorp Vault Integration - echo fix

### PR DESCRIPTION
**Pull request by vitalied**

### CODE REVIEW

File: lib/travis/build/appliances/setup_filter.rb [hunk 1]
```diff
@@ -105,7 +105,7 @@ def exports
           end
 
           def secrets
-            @secrets ||= env_secrets.concat(data.secrets).uniq
+            @secrets ||= env_secrets.concat(data.secrets).concat(data.vault_secrets).uniq
```
Comment (GbArc) [inline lib/travis/build/appliances/setup_filter.rb:108]: any chance data.vault_secrets will be nil? or the vault service runs always?

**Review by GbArc — APPROVED** [review]

File: lib/travis/build/appliances/setup_filter.rb (hunk 1 above)
Comment (vitalied) [reply lib/travis/build/appliances/setup_filter.rb:108]: > any chance data.vault_secrets will be nil? or the vault service runs always?

```
def vault_secrets
  Array(data[:vault_secrets])
end
```
```
Array(nil) => []
```

File: lib/travis/services/vault/keys/resolver.rb [hunk 2]
```diff
@@ -36,7 +35,7 @@ def call
             end
           end
 
-          data.vault_secrets = vault_secrets
+          data.vault_secrets = vault_secrets if vault_secrets.present?
```
Comment (vitalied) [inline lib/travis/services/vault/keys/resolver.rb:38]: @makemp `data.vault_secrets = vault_secrets.uniq if vault_secrets.present?`

**Review by vitalied — COMMENTED** [review]
pls fix

**Pull request merged**